node/builder.go

```diff
@@ -563,6 +570,17 @@ func ConfigStorageMiner(c interface{}) Option {
 		return Error(xerrors.Errorf("invalid config from repo, got: %T", c))
 	}
 
+	pricingConfig := cfg.Dealmaking.RetrievalPricing
+	if pricingConfig.Strategy == config.ExternalRetrievalPricing {
+		if pricingConfig.External == nil {
+			return Error(xerrors.New("retrieval pricing policy has been to set to external but external policy config is nil"))
+		}
+
+		if pricingConfig.External.Path == "" {
+			return Error(xerrors.New("retrieval pricing policy has been to set to external but external script path is empty"))
+		}
+	}
```

Comment:
I'd do `elseif Strategy != default -> error "only default or external are valid values"`

Reply:
@aarshkshah1992 I think you addressed this, but could you comment and mark as resolved?

Reply:
This is done.